lbrynet/core/Wallet.py

```diff
@@ -576,33 +813,57 @@ def _handle_claim_result(self, results):
             results['hex'] = claim_hex
             results['value'] = claim_dict
 
-        log.info("get claim info lbry://%s#%s", results['name'], results['claim_id'])
+        log.info("resolved lbry://%s#%s", results['name'], results['claim_id'])
         defer.returnValue(results)
 
     @defer.inlineCallbacks
-    def resolve_uri(self, uri):
-        resolve_results = yield self._get_value_for_uri(uri)
-        if 'claim' in resolve_results:
-            formatted = yield self._handle_claim_result(resolve_results)
-            resolve_results['claim'] = formatted
-            result = resolve_results
-        elif 'claims_in_channel' in resolve_results:
-            claims_for_return = []
-            for claim in resolve_results['claims_in_channel']:
-                formatted = yield self._handle_claim_result(claim)
-                claims_for_return.append(formatted)
-            resolve_results['claims_in_channel'] = claims_for_return
-            result = resolve_results
-        elif 'error' in resolve_results:
-            raise Exception(resolve_results['error'])
+    def resolve_uri(self, uri, check_cache=True):
+        cached_claim = yield self._storage.get_cached_claim_for_uri(uri, check_cache)
+        if cached_claim:
+            result = cached_claim
```

Comment:
just return the result here. there's no need to put everything else into an `else` block. returning early makes your code more readable